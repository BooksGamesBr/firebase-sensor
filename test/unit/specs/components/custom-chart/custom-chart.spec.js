import { shallow, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import CustomChartPage from '@/components/custom-chart/custom-chart-page'
import { FakeRooms, FakeCategories } from '../../../data/fake-data'

const localVue = createLocalVue()

localVue.use(Vuex)

describe('CustomChartPage.vue', () => {
  let getters
  let store

  beforeEach(() => {
    getters = {
      rooms: () => FakeRooms,
      categoriesSorted: () => FakeCategories
    }

    store = new Vuex.Store({
      getters
    })
  })

  it('shows a page', () => {
    const wrapper = shallow(CustomChartPage, {
      store,
      localVue
    })
    expect(wrapper.vm.rooms).toHaveLength(1)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
