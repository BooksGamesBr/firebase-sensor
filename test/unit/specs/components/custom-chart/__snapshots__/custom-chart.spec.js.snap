// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomChartPage.vue shows a page 1`] = `
<div>
    <div class="row">
        <div class="col-xs-12 col-sm-6 col-md-6">
            <select class="form-control">
                <option value="[object Object]">Temperature</option>
                <option value="[object Object]">Humidity</option>
            </select>
            <select class="form-control">
                <option value="24">24 hours</option>
                <option value="48">48 hours</option>
                <option value="168">1 week</option>
            </select>
        </div>
    </div>
    <div class="row">
        <div class="col-lg-12">
            <!---->
        </div>
    </div>
</div>
`;
